### ISSUE TITLE
index order in match_events output

### ISSUE THREAD

**Issue by justinsalamon** [craffel/mir_eval#171]
@bmcfee  I'm basing ```mir_eval.util.match_notes``` in on ```match_events```, and I think I have it down. One thing I noticed however is that calling ```matching = sorted(_bipartite_match(G).items())``` will return a list of tuples, where the value in each tuple is the index of the estimate node, and the second is the index of the reference node. Given that the order of the input parameters to the method is reference then estimate, I thought it would be more consistent for the returned tuples to be (ref_ind, est_ind) rather than (est_ind, ref_ind), as it is right now. This can be accomplished by flipping the graph construction:
```python
G = {}
    for ref_i, est_i in zip(*hits):
        if est_i not in G:
            G[est_i] = []
        G[est_i].append(ref_i)
```
(where I just swapped every occurrence of ```ref_i``` with ```est_i``` and vice-versa). 
I have two questions though:
1. Is _bipartite_match() symmetric? (I'd think so)
2. Does the internal structure of the returned ```matching``` list matter for anything else downstream? I know that for matching beats you just check for the size of the list, but not sure if it's used anywhere else.

Basically this is a tiny thing, but since I'm implementing ```match_notes``` now, I thought I'd ping you.

**Comment by bmcfee** [comment]
Yeah, that sounds reasonable.

1. Yes, since the edges are undirected, everything here is symmetric, so swapping ref and est here ought to be fine.
2. I don't think we use it anywhere else (it also gets called in segments, and conceivably anywhere else you have a matching problem), but always in terms of size.  I could imagine use-cases in which you'd want to preserve the matching structure -- eg, visualization/debugging/etc -- as opposed to just the count summary.

**Comment by justinsalamon** [comment]
ok cool. So I'll keep the flipped version in ```match_notes```, you wanna flip it in ```match_events```?
p.s - there's no unit test for ```match_events``` :ghost:

**Comment by craffel** [comment]
> p.s - there's no unit test for match_events :ghost:

There's a [unit test for _bipartite_match](https://github.com/craffel/mir_eval/blob/master/tests/test_util.py#L126), and the rest of `match_events` (which isn't much) gets hit pretty thoroughly by the rest of the tests.

**Comment by justinsalamon** [comment]
Fair enough. Btw, I just noticed that the docstring in ``match_events`` describes ``matching`` as: ``matching[i] == (i, j)`` where ``ref[i]`` matches ``est[j]``, which will only be correct when the graph construction is flipped as discussed above.

**Comment by bmcfee** [comment]
> Fair enough. Btw, I just noticed that the docstring in match_events describes matching as:
> "matching[i] == (i, j) where ref[i] matches est[j]"
> which will only be correct when the graph construction is flipped as discussed above.

Hah, derp.  If you want to correct all of that in this PR, that would be okay by me.

**Comment by justinsalamon** [comment]
roger that

**Comment by justinsalamon** [comment]
Fixed by #170
